planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp

```diff
@@ -1244,6 +1291,92 @@ bool NormalLaneChange::check_candidate_path_safety(
+      return std::nullopt;
+    }
+  } catch (const std::exception & e) {
+    RCLCPP_DEBUG(logger_, "failed to get terminal LC path: %s", e.what());
+    return std::nullopt;
+  }
+
+  // t = 2 * d / (v1 + v2)
+  const auto duration_to_lc_start =
+    2.0 * dist_to_terminal_start / (current_velocity + min_lc_velocity);
+  const auto lon_accel = std::invoke([&]() -> double {
+    if (duration_to_lc_start < calculation::eps) {
+      return 0.0;
+    }
+    return std::clamp(
+      (min_lc_velocity - current_velocity) / duration_to_lc_start,
+      lane_change_parameters_->trajectory.min_longitudinal_acc,
+      lane_change_parameters_->trajectory.max_longitudinal_acc);
+  });
+  const auto vel_on_prep = current_velocity + lon_accel * duration_to_lc_start;
+  const LaneChangePhaseMetrics prep_metric(
+    duration_to_lc_start, dist_to_terminal_start, vel_on_prep, lon_accel, lon_accel, 0.0);
+
+  if (terminal_lane_change_path_) {
+    terminal_lane_change_path_->info.set_prepare(prep_metric);
+    if (prepare_segment.points.empty()) {
+      terminal_lane_change_path_->path = terminal_lane_change_path_->shifted_path.path;
+      return terminal_lane_change_path_->path;
+    }
+    prepare_segment.points.pop_back();
+    terminal_lane_change_path_->path =
+      utils::combinePath(prepare_segment, terminal_lane_change_path_->shifted_path.path);
+    return terminal_lane_change_path_->path;
+  }
+
+  const auto & lane_changing_start_pose = prepare_segment.points.back().point.pose;
+  const auto & target_lanes = common_data_ptr_->lanes_ptr->target;
+  const auto shift_length =
+    lanelet::utils::getLateralDistanceToClosestLanelet(target_lanes, lane_changing_start_pose);
+
+  const auto dist_lc_start_to_end_of_lanes = calculation::calc_dist_from_pose_to_terminal_end(
+    common_data_ptr_, common_data_ptr_->lanes_ptr->target_neighbor,
+    prepare_segment.points.back().point.pose);
+
+  const auto max_lane_changing_length = std::invoke([&]() {
+    double max_length = transient_data.dist_to_terminal_end - prep_metric.length;
+    max_length =
+      std::min(max_length, dist_lc_start_to_end_of_lanes - transient_data.next_dist_buffer.min);
+    return max_length;
+  });
+
+  const auto max_path_velocity = prepare_segment.points.back().point.longitudinal_velocity_mps;
+  constexpr double lane_changing_lon_accel{0.0};
+  const auto lane_changing_metrics = calculation::calc_shift_phase_metrics(
+    common_data_ptr_, shift_length, prep_metric.velocity, max_path_velocity,
+    lane_changing_lon_accel, max_lane_changing_length);
+
+  const auto sorted_lane_ids = utils::lane_change::get_sorted_lane_ids(common_data_ptr_);
+
+  LaneChangePath candidate_path;
+  for (const auto & lc_metric : lane_changing_metrics) {
+    try {
+      candidate_path = get_candidate_path(
+        prep_metric, lc_metric, prepare_segment, sorted_lane_ids, lane_changing_start_pose,
+        shift_length);
+    } catch (const std::exception & e) {
+      continue;
```

Comment:
consider printing something.

Reply:
since I'm working on separate PR for debugging improvement, I will leave it as is for now